From lkml@rtr.ca  Mon Jan  5 16:10:37 2009
From: Mark Lord <lkml@rtr.ca>
Date: Fri, 02 Jan 2009 02:48:24 -0500
Subject: USB: fix minor nit in usbfs checking
To: Greg KH <gregkh@suse.de>
Message-ID: <495DC6C8.7070501@rtr.ca>

From: Mark Lord <lkml@rtr.ca>

One minor nit did show up, though.  The patch below
seems to make more sense than the code does without it.

Signed-off-by: Mark Lord <mlord@pobox.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/core/devio.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/devio.c
+++ b/drivers/usb/core/devio.c
@@ -1317,7 +1317,7 @@ static int get_urb32(struct usbdevfs_urb
 	if (__get_user(uptr, &uurb->buffer))
 		return -EFAULT;
 	kurb->buffer = compat_ptr(uptr);
-	if (__get_user(uptr, &uurb->buffer))
+	if (__get_user(uptr, &uurb->usercontext))
 		return -EFAULT;
 	kurb->usercontext = compat_ptr(uptr);
 
